From e6a2e71831f0699e9759e0579a00eea5fcb93696 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Wed, 24 Apr 2019 20:23:00 +0200
Subject: [PATCH] FindXml2: use pkg-config to find libxml2 dependencies

On UNIX, when statically built, libxml2 can depends on other libraries such
as lzma. These dependencies are already retrieved through pkg-config so
append them to LIBXML2_LIBRARIES otherwise static build will fail

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
[Upstream status: not sent yet]
---
 cmake/modules/FindLibXml2.cmake | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/cmake/modules/FindLibXml2.cmake b/cmake/modules/FindLibXml2.cmake
index 99d70af57d..2a4b1f470f 100644
--- a/cmake/modules/FindLibXml2.cmake
+++ b/cmake/modules/FindLibXml2.cmake
@@ -98,6 +98,10 @@ endif()
 
 set(LIBXML2_INCLUDE_DIRS ${LIBXML2_INCLUDE_DIR} ${PC_LIBXML_INCLUDE_DIRS})
 set(LIBXML2_LIBRARIES ${LIBXML2_LIBRARY})
+# Include transitive dependencies for static linking.
+if(UNIX)
+    list(APPEND LIBXML2_LIBRARIES ${PC_LIBXML_LIBRARIES})
+endif()
 
 include(FindPackageHandleStandardArgs)
 FIND_PACKAGE_HANDLE_STANDARD_ARGS(LibXml2
-- 
2.20.1

